servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/OnErrorResumeCompletable.java

```diff
@@ -89,7 +89,6 @@ public void onError(Throwable throwable) {
             try {
                 next = predicate.test(throwable) ? requireNonNull(nextFactory.apply(throwable)) : null;
             } catch (Throwable t) {
-                t.addSuppressed(throwable);
```

Comment:
Would the supressed be likely to have a different stack root? ie. from another thread with more context? It seems fairly low cost to keep it around unless the suppressed exception is truly redundant.

Reply:
I will revert this change.

I think we need to do a more general investigation into `addSuppressed` usage which has the following potential issues:
- we don't use `addSuppressed` consistently
- it creates memory leaks if static/shared exceptions are thrown
- restrictions on usage such as self-add result in throwing which breaks async control flow